gensim/models/word2vec.py

```diff
@@ -1819,18 +1820,31 @@ def predict_output_word(self, context_words_list, topn=10):
         if not hasattr(self.wv, 'vectors') or not hasattr(self, 'syn1neg'):
             raise RuntimeError("Parameters required for predicting the output words not found.")
 
-        word2_indices = [self.wv.get_index(w) for w in context_words_list if w in self.wv]
-        if not word2_indices:
-            logger.warning("All the input context words are out-of-vocabulary for the current model.")
-            return None
+        if all(isinstance(w, int) for w in context_words_list):
+            # then, indices were passed. Check they are valid
+            word2_indices = np.array(context_words_list)
+            if np.any(word2_indices < 0):
+                logger.warning("All input context word indices must be non-negative.")
```

Comment:
For consistency, why not ignore `<= 0` indices instead? Same as you ignore `>= vectors.shape[0]` indices below.

Reply:
You're completely right. I wrote it thinking "let's do it this way to be consistent with the existing code", but it is actually not. So we just remove lines 1826 to 1828? Then there will only be selection of valid indices, and then checking if there remain some.